apps/extension/src/hooks/latest-block-height.ts

```diff
@@ -1,106 +1,63 @@
 import { createGrpcWebTransport } from '@connectrpc/connect-web';
 import { TendermintProxyService } from '@penumbra-zone/protobuf';
 import { ChainRegistryClient } from '@penumbra-labs/registry';
 import { useStore } from '../state';
 import { networkSelector } from '../state/network';
 
-// Utility function to fetch the block height by randomly querying one of the RPC endpoints
-// from the chain registry, using a recursive callback to try another endpoint if the current
-// one fails. Additionally, this implements a timeout mechanism at the request level to avoid
-// hanging from stalled requests.
-const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
-  if (endpoints.length === 0) {
-    throw new Error('All RPC endpoints failed to fetch the block height.');
-  }
-
-  // Randomly select an RPC endpoint from the chain registry
-  const randomGrpcEndpoint = sample(endpoints);
-  if (!randomGrpcEndpoint) {
-    throw new Error('No RPC endpoints found.');
-  }
-
-  try {
-    return await fetchBlockHeightWithTimeout(randomGrpcEndpoint);
-  } catch (e) {
-    // Remove the current endpoint from the list and retry with remaining endpoints
-    const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomGrpcEndpoint);
-    return fetchBlockHeightWithFallback(remainingEndpoints);
-  }
-};
-
-// Fetch the block height from a specific RPC endpoint with a request-level timeout that superceeds
-// the channel transport-level timeout to prevent hanging requests.
-export const fetchBlockHeightWithTimeout = async (
-  grpcEndpoint: string,
-  timeoutMs = 5000,
-): Promise<number> => {
-  return new Promise((resolve, reject) => {
-    const timeout = setTimeout(() => {
-      reject(new Error('Request timed out'));
-    }, timeoutMs);
-
+const fetchRemoteLatestBlockHeight = async (rpcUrls: string[], timeoutMs = 5000) => {
+  for (const baseUrl of rpcUrls) {
     const tendermintClient = createPromiseClient(
       TendermintProxyService,
-      createGrpcWebTransport({ baseUrl: grpcEndpoint }),
+      createGrpcWebTransport({ baseUrl }),
     );
 
-    tendermintClient
-      .getStatus({})
-      .then(result => {
-        if (!result.syncInfo) {
-          reject(new Error('No syncInfo in getStatus result'));
-        }
-        clearTimeout(timeout);
-        resolve(Number(result.syncInfo?.latestBlockHeight));
-      })
-      .catch(() => {
-        clearTimeout(timeout);
-        reject(new Error('RPC request timed out while fetching block height'));
-      });
-  });
-};
-
-// Fetch the block height from a specific RPC endpoint.
-export const fetchBlockHeight = async (grpcEndpoint: string): Promise<number> => {
-  const tendermintClient = createPromiseClient(
-    TendermintProxyService,
-    createGrpcWebTransport({ baseUrl: grpcEndpoint }),
-  );
+    const latestBlockHeight = await tendermintClient
+      .getStatus({}, { signal: AbortSignal.timeout(timeoutMs) })
```

Comment:
you could also use `timeoutMs` directly

```suggestion
      .getStatus({}, { timeoutMs })
```

this is fully unfounded, but i wonder if using the `timeoutMs` option might actually wait for the request to connect before starting the timer. a `signal` will definitely abort the request without any conditions.